autorest/azure/token.go

```diff
@@ -104,6 +154,79 @@ func (spt *ServicePrincipalToken) EnsureFresh() error {
 
 // Refresh obtains a fresh token for the Service Principal.
 func (spt *ServicePrincipalToken) Refresh() error {
+	newToken, err := spt.GetOauthToken()
+	if err != nil {
+		return err
+	}
+	spt.Token = *newToken
+
+	return nil
+}
+
+// SetAuthValues is a method of the interface ServicePrincipalTokenSecret
+// It will populate the form submitted during oAuth Token Acquisition using a static client_secret
+func (tokenSecret *ServicePrincipalTokenSecret) SetAuthValues(spt *ServicePrincipalToken, v *url.Values) error {
```

Comment:
I strongly prefer methods defined on a struct to be kept adjacent to the struct itself.